### PR TITLE
fix(node): fix end point in node range

### PR DESCRIPTION
**Pull request by zoltan-kski**
Hello,

This PR fixes a bug in `Node.range()` and adds related test.

Let me know if something else is needed for this PR.

### CODE REVIEW

File: src/test.zig [hunk 1]
```diff
@@ -203,6 +203,11 @@ test "Node" {
     try testing.expectEqual(0, node.startPoint().column);
     try testing.expectEqual(13, node.endPoint().column);
 
+    try testing.expectEqual(node.startByte(), node.range().start_byte);
+    try testing.expectEqual(node.endByte(), node.range().end_byte);
+    try testing.expectEqual(node.startPoint(), node.range().start_point);
+    try testing.expectEqual(node.endPoint(), node.range().end_point);
```
Comment (ObserverOfTime) [inline src/test.zig:209]:
```suggestion
    var range = node.range();
    try testing.expectEqual(0, range.start_byte);
    try testing.expectEqual(13, range.end_byte);
    try testing.expectEqual(0, range.start_point);
    try testing.expectEqual(13, range.end_point);
```

Comment (zoltan-kski) [reply src/test.zig]: @ObserverOfTime thanks for the review, I updated in the test in dece4e50a2d20726aaab2c0e38eba97cb97623f6.

File: src/test.zig [hunk 2]
```diff
@@ -203,6 +203,12 @@ test "Node" {
     try testing.expectEqual(0, node.startPoint().column);
     try testing.expectEqual(13, node.endPoint().column);
 
+    const range = node.range();
+    try testing.expectEqual(0, range.start_byte);
+    try testing.expectEqual(13, range.end_byte);
+    try testing.expectEqual(ts.Point{.row = 0, .column = 0}, range.start_point);
+    try testing.expectEqual(ts.Point{.row = 0, .column = 13}, range.end_point);
```
Comment (ObserverOfTime) [inline src/test.zig:210]:
```suggestion
    try testing.expectEqual(0, range.start_point.column);
    try testing.expectEqual(13, range.end_point.column);
```

**Comment by ObserverOfTime** [comment]
Thank you.

**Pull request merged**